geomstats/_backend/pytorch/__init__.py

```diff
@@ -804,3 +804,15 @@ def mat_from_diag_triu_tril(diag, tri_upp, tri_low):
     mat[..., j, k] = tri_upp
     mat[..., k, j] = tri_low
     return mat
+
+
+def vec_to_sym(vec, mat_dim):
+    shape = (mat_dim, mat_dim)  
+    shape = (vec.shape[0],) + shape if vec.ndim>1 else shape
+    mat = torch.zeros(shape)
+    i, i = gs.diag_indices(mat_dim)
+    j, k = gs.tril_indices(mat_dim)
+    mat[... , j, k] = vec
```

Comment:
will this be differentiable by automatic differentiation?
This is why the former methods in SymmetricMatrix were written without assignment.

Reply:
@nguigs  
It works actually. pytorch is smart. The reason is twofold.

indexes that the diag_indices returns is int64 torch tensors and importantly
There is differentiable operation called <IndexPutBackward> (whose gradient is straight-through). Here is the working example
```
ind, = diag_indices(3, ndim =1)
mat = torch.zeros((3,3))
diag = torch.tensor([2., 3., 4.], requires_grad=True)
squared_diag = diag**2
mat[ind,ind] = squared_diag
sum = mat.sum()
sum.backward()
print(diag.grad)

```

now grad tensor should be 2*diag (we have squared and added) if everything is correct. output is indeed tensor([4., 6., 8.])